symengine/test_visitors.h

```diff
@@ -400,6 +400,31 @@ class RationalVisitor : public BaseVisitor<RationalVisitor>
 
     tribool apply(const Basic &b);
 };
+
+class AlgebraicVisitor : public BaseVisitor<AlgebraicVisitor>
+{
+private:
+    tribool is_algebraic_;
+    const Assumptions *assumptions_;
+
+    void error();
+
+public:
+    AlgebraicVisitor(const Assumptions *assumptions) : assumptions_(assumptions)
+    {
+    }
+
+    void bvisit(const Basic &x);
+    void bvisit(const Integer &x);
+    void bvisit(const Rational &x);
+    void bvisit(const Set &x);
+    void bvisit(const Relational &x);
+    void bvisit(const Boolean &x);
+    void bvisit(const Constant &x);
```

Comment:
Should we be implementing things like Add, Mul, Pow?

Reply:
As far as I understand there are no general rules known for these. We'd have to create special cases for what is know.

Reply:
algebraic + algebraic = algebraic, right? For Pow, we'll have to have special cases like https://github.com/sympy/sympy/blob/ef3d570ac8f9b76855351bb712f8dbe2f1b2c69a/sympy/core/power.py#L1383

Reply:
grepping `_eval_is_algebraic` in sympy will give you all the rules that sympy implements.